daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -69,27 +70,32 @@ private[lf] object PartialTransaction {
     */
   final case class Context(
       info: ContextInfo,
+      minChildVersion: TxVersion,
       children: BackStack[NodeId],
       nextActionChildIdx: Int,
   ) {
-    def addActionChild(child: NodeId): Context =
-      Context(info, children :+ child, nextActionChildIdx + 1)
+    def addActionChild(child: NodeId, version: TxVersion): Context = {
+      Context(info, minChildVersion min version, children :+ child, nextActionChildIdx + 1)
+    }
     def addRollbackChild(child: NodeId, nextActionChildIdx: Int): Context =
-      Context(info, children :+ child, nextActionChildIdx)
+      Context(info, minChildVersion, children :+ child, nextActionChildIdx)
```

Comment:
can we propagate the version from the rollback node? I know it’s not needed since we ensure that they are not older but imho it makes the semantics of what `minChildVersion` records much clearer.

Reply:
ok

Reply:
done